doc/source/guide/derivatives_guide.rst

```diff
@@ -0,0 +1,151 @@
+.. _derivatives_in_depth:
+
+##############################################
+A guide to the different notions of derivative
+##############################################
+
+The concept of a derivative is one of the core concepts of mathematical analysis and is used where we want a linear approximation of an arbitrary function. Since the derivative is used in different contexts to mean slightly different things, this guide has been written to clarify and introduce what the different concepts mean.
+
+In short, different notions of derivatives that will be discussed here are:
+
+* **Derivative**. When we write derivative in ODL code and documentation, we intend the derivative of a `Operator` :math:`A : \mathcal{X} \rightarrow \mathcal{Y}` w.r.t to a disturbance in :math:`x`, i.e a linear approximation of :math:`A(x + h)` for small :math:`h`. The derivative in a point :math:`x` is an operator :math:`A'(x) : \mathcal{X} \rightarrow \mathcal{Y}`.
```

Comment:
Or more clearly, match the column of the first non-space character after the first bullet :-)